site/app/models/gradeable/AutogradingTestcase.php

```diff
@@ -36,6 +36,12 @@ class AutogradingTestcase extends AbstractModel {
     protected $view_testcase_message = true;
     /** @prop @var string */
     protected $testcase_label = '';
+    /** @prop @var array */
+    protected $dispatcher_actions = array();
+    /** @prop @var array */
+    protected $graphics_actions = array();
+    /** @prop @var boolean */
+    protected $publish_actions = false;
```

Comment:
`@prop-read` accomplishes the same thing as adding the `set` functions that throw. See https://submitty.org/developer/developing_the_php_site/models.

Reply:
Also, please add a `@method getDispatcherActions` et al. above the class as that is used by phpstan for type hinting wherever the functions are called.